src/packages/core/collection/collection.context.ts

```diff
@@ -30,107 +29,119 @@ export class UmbCollectionContext<ItemType, FilterModelType extends UmbCollectio
 	#filter = new UmbObjectState<FilterModelType | object>({});
 	public readonly filter = this.#filter.asObservable();
 
-	repository?: UmbCollectionRepository;
+	#views = new UmbArrayState<ManifestCollectionView>([]);
+	public readonly views = this.#views.asObservable();
 
-	/*
-	TODO:
-	private _search = new StringState('');
-	public readonly search = this._search.asObservable();
-	*/
+	#currentView = new UmbObjectState<ManifestCollectionView | undefined>(undefined);
+	public readonly currentView = this.#currentView.asObservable();
+
+	repository?: UmbCollectionRepository;
+	collectionRootPathname: string;
 
 	constructor(host: UmbControllerHostElement, entityType: string, repositoryAlias: string) {
-		this._entityType = entityType;
-		this._host = host;
-
-		new UmbObserverController(
-			this._host,
-			umbExtensionsRegistry.getByTypeAndAlias('repository', repositoryAlias),
-			async (repositoryManifest) => {
-				if (repositoryManifest) {
-					const result = await createExtensionApi(repositoryManifest, [this._host]);
-					this.repository = result as UmbCollectionRepository;
-					this._onRepositoryReady();
-				}
-			}
-		);
+		this.entityType = entityType;
+		this.host = host;
+
+		const currentUrl = new URL(window.location.href);
+		this.collectionRootPathname = currentUrl.pathname.substring(0, currentUrl.pathname.lastIndexOf('/'));
+
+		this.init = Promise.all([
+			new UmbObserverController(
+				this.host,
+				umbExtensionsRegistry.getByTypeAndAlias('repository', repositoryAlias),
+				async (repositoryManifest) => {
+					if (repositoryManifest) {
+						const result = await createExtensionApi(repositoryManifest, [this.host]);
+						this.repository = result as UmbCollectionRepository;
+						this.requestCollection();
+					}
+				},
+			).asPromise(),
+
+			new UmbObserverController(
+				this.host,
+				umbExtensionsRegistry.extensionsOfType('collectionView').pipe(
+					map((extensions) => {
+						return extensions.filter((extension) => extension.conditions.entityType === this.getEntityType());
+					}),
+				),
+				(views) => {
+					this.#views.next(views);
+					this.#setCurrentView();
+				},
+			).asPromise(),
+		]);
 	}
 
+	/**
+	 * Returns true if the given id is selected.
+	 * @param {string} id
+	 * @return {Boolean}
+	 * @memberof UmbCollectionContext
+	 */
 	public isSelected(id: string) {
 		return this.#selection.getValue().includes(id);
 	}
 
-	public setSelection(value: Array<string>) {
-		if (!value) return;
-		this.#selection.next(value);
+	/**
+	 * Sets the current selection.
+	 * @param {Array<string>} selection
+	 * @memberof UmbCollectionContext
+	 */
+	public setSelection(selection: Array<string>) {
```

Comment:
We do have a `UmbSelectionManagerBase`, could it make sense to use that as part of this context. So we instantiate that on a prop called `selection` ?  Just an idea, if you think it could make sense?

Reply:
I have updated the code so all logic now comes from the SelectionManager. I can't figure out what the best practice is. Should we talk directly with the selection manager or implement methods in the class that talk to the selection manager? I have currently gone with the last solution because then each implementation can pick and choose the features. But it also adds a lot more code. 

Any good ideas?